server.js

```diff
@@ -44,18 +72,8 @@ function handleSocketConnection(socket) {
     io.to(currentRoom).emit('lobbyUpdate', users.list())
 
     if (users.count() >= MIN_PLAYER_COUNT) {
-      console.log(`${currentRoom} will start a game in ${START_GAME_DELAY} sec`)
-      clearTimeout(countDownTimer)
-      countDownTimer = setTimeout(function startAGame() {
-        if (users.count() >= MIN_PLAYER_COUNT) { // if nobody has left lobby
-          io.to(currentRoom).emit('stateChange', 'game')
-          console.log(`${currentRoom} started a game!`)
-          currentRoom = generateRoomName()
-          users.clear()
-          io.emit('lobbyUpdate', users.list())
-        }
-      }, START_GAME_DELAY * ONE_SECOND + DELAY_BUFFER_IN_MS)
-      io.to(currentRoom).emit('countdown', START_GAME_DELAY)
+      startGame()
+      assignRoles(currentRoom)
```

Comment:
This is in the wrong place. We only want to assign roles after the game state event has been emitted, so it should be executed some time right after a successful countdown and a start of game